OpenRA.Mods.Common/Traits/Render/WithVoxelBarrel.cs

```diff
@@ -43,13 +45,13 @@ public IEnumerable<VoxelAnimation> RenderPreviewVoxels(ActorPreviewInitializer i
 			var voxel = VoxelProvider.GetVoxel(image, Sequence);
 
 			var turretFacing = Turreted.GetInitialTurretFacing(init, t.InitialFacing, t.Turret);
-			var turretOrientation = body.QuantizeOrientation(new WRot(WAngle.Zero, WAngle.Zero, WAngle.FromFacing(turretFacing) - orientation.Yaw), facings);
+			var turretOrientation = body.QuantizeOrientation(new WRot(WAngle.Zero, WAngle.Zero, WAngle.FromFacing(turretFacing) - orientation().Yaw), facings);
 
 			var quantizedTurret = body.QuantizeOrientation(turretOrientation, facings);
-			var quantizedBody = body.QuantizeOrientation(orientation, facings);
+			var quantizedBody = body.QuantizeOrientation(orientation(), facings);
```

Comment:
This will need to be moved inside the `() => barrelOffset`, otherwise i will retain its value from when the unit was picked up.